static/js/message_edit.js

```diff
@@ -904,56 +906,57 @@ export function edit_last_sent_message() {
     });
 }
 
-function hide_delete_btn_show_spinner(deleting) {
-    if (deleting) {
-        $("do_delete_message_button").prop("disabled", true);
-        $("#delete_message_modal > div.modal-footer > button").hide();
-        const delete_spinner = $("#do_delete_message_spinner");
-        loading.make_indicator(delete_spinner, {abs_positioned: true});
-    } else {
-        loading.destroy_indicator($("#do_delete_message_spinner"));
-        $("#do_delete_message_button").prop("disabled", false);
-        $("#delete_message_modal > div.modal-footer > button").show();
-    }
-}
-
 export function delete_message(msg_id) {
     $("#delete-message-error").html("");
-    overlays.open_modal("#delete_message_modal");
+
+    const html_body = render_delete_message_modal_body();
+    const modal_parent = $("#message_feed_container");
+
+    function do_delete_message() {
+        confirm_dialog.show_confirm_dialog_spinner();
+
+        currently_deleting_messages.push(msg_id);
+        channel.del({
+            url: "/json/messages/" + msg_id,
+            success() {
+                currently_deleting_messages = currently_deleting_messages.filter(
+                    (id) => id !== msg_id,
+                );
+                confirm_dialog.hide_confirm_dialog_spinner();
+                overlays.close_modal("#confirm_dialog_modal");
+            },
+            error(xhr) {
+                currently_deleting_messages = currently_deleting_messages.filter(
+                    (id) => id !== msg_id,
+                );
+
+                confirm_dialog.hide_confirm_dialog_spinner();
+                overlays.close_modal("#confirm_dialog_modal");
+                ui_report.error(
+                    $t_html({defaultMessage: "Error deleting message"}),
+                    xhr,
+                    $("#delete-message-error"),
+                );
+            },
+        });
+    }
+
+    confirm_dialog.launch({
+        parent: modal_parent,
+        html_heading: $t_html({defaultMessage: "Delete message"}),
+        html_body,
+        html_yes_button: $t_html({defaultMessage: "Confirm"}),
+        help_link: "/help/edit-or-delete-a-message#delete-a-message",
+        on_click: do_delete_message,
+        fade: false,
+        loading_spinner: true,
+    });
+
     if (currently_deleting_messages.includes(msg_id)) {
-        hide_delete_btn_show_spinner(true);
+        confirm_dialog.show_confirm_dialog_spinner();
     } else {
-        hide_delete_btn_show_spinner(false);
+        confirm_dialog.hide_confirm_dialog_spinner();
     }
```

Comment:
Do you understand the purpose of this block?  It doesn't make any sense to me and I suspect should be just deleted.

(Won't it always be false?)

Reply:
The way this used to work was that, since it's always false, the spinner is removed by `loading.destroy_indicator` which [added](https://github.com/zulip/zulip/blob/386ddf15178ea1bec70338c37c7815b9f26cabfc/static/js/loading.js#L64) `width: 0` to the spinner which in turn caused the spinner to be centered by `margin: 0 auto` [here](https://github.com/zulip/zulip/blob/386ddf15178ea1bec70338c37c7815b9f26cabfc/static/styles/zulip.css#L2343).
I was going to remove this block after changing the CSS above. I will do it now.